src/main/java/org/spongepowered/api/text/placeholder/PlaceholderText.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.placeholder;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.TextRepresentable;
+import org.spongepowered.api.util.ResettableBuilder;
+
+/**
+ * A {@link TextRepresentable} that can be used in {@link Text} building methods
+ * that represents a placeholder in text.
+ *
+ * <p>A {@link PlaceholderText} is the collection of a {@link PlaceholderParser}
+ * along with contextual data in the supplied {@link PlaceholderContext},
+ * enabling its use in a {@link Text} object.</p>
+ *
+ * <p>While such placeholders will generally be built from tokenised strings,
+ * these objects make no assumption about the format of text templating. Such a
+ * system can therefore be used by other templating engines without conforming
+ * to a particular standard.</p>
+ */
+public interface PlaceholderText extends TextRepresentable {
```

Comment:
Maybe add a note, that the placeholder will be resolved when converting it to a `Text` and not during the creation of the `PlaceholderText` instance.